pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -520,6 +531,14 @@ func (plugin *kubenetNetworkPlugin) teardown(namespace string, name string, id k
 		}); err != nil {
 			errList = append(errList, err)
 		}
+		if err = plugin.hostportManagerv6.Remove(id.ID, &hostport.PodPortMapping{
```

Comment:
I was struggling with this, and I don't have clear the best approach. Let me try to share my thoughts:

With the Add() method we can choose the handler because we know the IP family beforehand.
However, with Remove() we receive a list of `portMappings` that can have IPv4 and IPv6 

https://github.com/kubernetes/kubernetes/blob/6d8dd211111474f277ef8e390a3eeed2465fb9e0/pkg/kubelet/dockershim/docker_service.go#L377

Also, Remove() doesn't take IP addresses on consideration, it finds the iptables chains based on the hash.

https://github.com/kubernetes/kubernetes/blob/a33840e0232aaf2b5c14c1c754d9189d7e841ec1/pkg/kubelet/dockershim/network/hostport/hostport_manager.go#L311

Maybe I should process the list of portMappings and create one list per IP family?

Reply:
PTAL